ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorMR.java

```diff
@@ -817,6 +816,7 @@ public void map(WritableComparable key, CompactorInputSplit split,
           deleteEventWriter.write(value);
           reporter.progress();
         } else {
+          getWriter(reporter, reader.getObjectInspector(), split.getBucket());
```

Comment:
could we refactor to:
````
        if (sawDeleteRecord && acidOperationalProperties.isSplitUpdate()) {
            getDeleteEventWriter(reporter, reader.getObjectInspector(), split.getBucket())
                .write(value);
          reporter.progress();
        } else {
          getWriter(reporter, reader.getObjectInspector(), split.getBucket())
              .write(value);
          reporter.progress();
        }
````

Reply:
`getWriter` and `getDeleteEventWriter` methods initializes `writer` object and `deleteEventWriter` object respectively which are initially null.

Reply:
Refactored. Done.